lib/LaTeXML/Package/TeX.pool.ltxml

```diff
@@ -5479,22 +5479,27 @@ DefMathI('\biguplus', undef, "\x{228E}",
   meaning   => 'symmetric-difference',
   mathstyle => \&doVariablesizeOp,
   font      => { size => 'Big' });
-DefConstructorI('\limits', undef, sub {
-    my $node = $_[0]->getElement;
-    $_[0]->setAttribute($_[0]->getLastChildElement($node) || $node, scriptpos => 'mid'); });
-DefConstructorI('\nolimits', undef, sub {
-    my $node = $_[0]->getElement;
-    $node = $_[0]->getLastChildElement($node) || $node;
-    $node->removeAttribute('scriptpos'); });    # default is 'post', so we can just remove the attrib.
-DefConstructorI('\displaylimits', undef, sub {
-    my ($document, %props) = @_;
-    my $node = $_[0]->getElement;
-    $node = $_[0]->getLastChildElement($node) || $node;
-    if (($props{mathstyle} || 'text') eq 'display') {
-      $document->setAttribute($node, scriptpos => 'mid'); }
-    else {
-      $node->removeAttribute('scriptpos'); } },
-  properties => sub { (mathstyle => LookupValue('font')->getMathstyle); });
+DefConstructorI('\limits', undef, '',
+  afterDigest => sub { mergeLimits('mid'); },
+  properties  => { isSpace => 1 });
+DefConstructorI('\nolimits', undef, '',
+  afterDigest => sub { mergeLimits('post'); },
+  properties  => { isSpace => 1 });
+DefConstructorI('\displaylimits', undef, '',
+  afterDigest => sub {
+    mergeLimits((($_[1]->getProperty('mathstyle') || '') eq 'display' ? 'mid' : 'post')); },
+  properties => { isSpace => 1 });
+
+sub mergeLimits {
+  my ($pos) = @_;
+  for (my $i = scalar(@LaTeXML::LIST) - 1 ; $i >= 0 ; $i--) {
+    my $box   = $LaTeXML::LIST[$i];
+    my $prev  = $box->getProperty('scriptpos') || '';
+    my $level = ($prev =~ /^\w*(\d+)$/ ? $1 : $STATE->getStomach->getScriptLevel || '');
```

Comment:
minor, but `$STATE->getStomach->getScriptLevel` could probably be called once outside of the loop and reused.

Reply:
the loop should typically be short anyway